test/inductor/test_distributed_patterns.py

```diff
@@ -13,76 +12,100 @@
 
 # Fake distributed
 WORLD_SIZE = 2
-RESIZE = True
 
 
-def init_fake_distributed():
+def init_fake_distributed(device="cpu"):
     @torch.no_grad
     def all_gather(t):
-        return torch.cat([t] * WORLD_SIZE, 0)
+        # clone since all_gather input and output should not be aliases.
+        return torch.cat([t] * WORLD_SIZE, 0).clone()
 
     @torch.no_grad
     def reduce_scatter(t):
-        return t.narrow(0, 0, t.size(0) // WORLD_SIZE)
+        # clone since reduce_scatter input and output should not be aliases.
+        return t.narrow(0, 0, t.size(0) // WORLD_SIZE).clone()
```

Comment:
ahh nice catch (and I guess this is more representative of FSDP2 - after initialization, the full params are freed and only the local shard / subset is in memory)